src/java/org/apache/cassandra/locator/AbstractReplicationStrategy.java

```diff
@@ -324,7 +324,8 @@ private static AbstractReplicationStrategy createInternal(String keyspaceName,
         try
         {
             Constructor<? extends AbstractReplicationStrategy> constructor = strategyClass.getConstructor(parameterTypes);
-            strategy = constructor.newInstance(keyspaceName, tokenMetadata, snitch, strategyOptions);
+            IEndpointSnitch endpointSnitch = snitch == null ? new SimpleSnitch() : snitch;
```

Comment:
This will propagate `null` snitch in client mode. I just have to put there something non-nullish.

Reply:
Agreed. I did this in https://github.com/apache/cassandra/commit/fecac728ec945e9b376a5fb2c4af54f337d744cf.